### PR TITLE
Move ensemble_calibration to sit beneath statistical_postprocessing

### PR DESCRIPTION
**Pull request by bayliffe**
Moves the EMOS (ensemble_calibration) code to sit within a directory called statistical_postprocessing. This will also contain reliability calibration.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/statistical_postprocessing/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  4
- improver_tests/statistical_postprocessing/utilities/test_utilities.py  2
- improver_tests/statistical_postprocessing/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  3
- improver_tests/statistical_postprocessing/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py  2
- improver/statistical_postprocessing/ensemble_calibration.py  13
         

Clones added
============
- improver_tests/statistical_postprocessing/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  12
- improver_tests/statistical_postprocessing/utilities/test_utilities.py  10
- improver_tests/statistical_postprocessing/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  6
- improver/statistical_postprocessing/ensemble_calibration.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4840541&bid=16072661)

**Comment by gavinevans** [comment]
Given Tomek's suggested renaming elsewhere, perhaps it would be best if from `improver.statistical_postprocessing.ensemble_calibration` became `improver.calibration.emos` with other techniques being added to the calibration directory. What do you think?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py  2
- improver/calibration/ensemble_calibration.py  13
- improver_tests/calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  3
- improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  4
- improver_tests/calibration/utilities/test_utilities.py  2
         

Clones added
============
- improver/calibration/ensemble_calibration.py  2
- improver_tests/calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  6
- improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  12
- improver_tests/calibration/utilities/test_utilities.py  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4840541&bid=16072661)

File: improver/ensemble_copula_coupling/ensemble_copula_coupling.py [hunk 1]
```diff
@@ -40,7 +40,8 @@
 from scipy import stats
 
 from improver import BasePlugin
-from improver.ensemble_calibration.utilities import convert_cube_data_to_2d
+from improver.calibration.utilities import (
```
Comment (fionaRust) [inline improver/ensemble_copula_coupling/ensemble_copula_coupling.py]: fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_EnsembleReordering.py [hunk 2]
```diff
@@ -44,9 +44,9 @@
     EnsembleReordering as Plugin
 from improver.utilities.warnings_handler import ManageWarnings
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period, set_up_cube,
-    set_up_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_EnsembleReordering.py]: Fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromMeanAndVariance.py [hunk 3]
```diff
@@ -44,9 +44,9 @@
     GeneratePercentilesFromMeanAndVariance as Plugin
 from improver.utilities.warnings_handler import ManageWarnings
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period,
-    set_up_spot_temperature_cube, set_up_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromMeanAndVariance.py]: Fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromProbabilities.py [hunk 4]
```diff
@@ -46,11 +46,11 @@
 from improver.metadata.probabilistic import find_threshold_coordinate
 from improver.utilities.warnings_handler import ManageWarnings
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period,
-    set_up_probability_above_threshold_spot_temperature_cube,
-    set_up_probability_above_threshold_temperature_cube,
-    set_up_probability_threshold_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromProbabilities.py]: fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GenerateProbabilitiesFromMeanAndVariance.py [hunk 5]
```diff
@@ -43,8 +43,8 @@
 from improver.metadata.probabilistic import find_threshold_coordinate
 from improver.utilities.cube_manipulation import enforce_coordinate_ordering
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    set_up_probability_above_threshold_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GenerateProbabilitiesFromMeanAndVariance.py]: fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_RebadgePercentilesAsMembers.py [hunk 6]
```diff
@@ -44,8 +44,9 @@
 from improver.ensemble_copula_coupling.ensemble_copula_coupling import \
     RebadgePercentilesAsRealizations as Plugin
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period, set_up_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_RebadgePercentilesAsMembers.py]: fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_ResamplePercentiles.py [hunk 7]
```diff
@@ -41,9 +41,9 @@
     ResamplePercentiles as Plugin
 from improver.utilities.warnings_handler import ManageWarnings
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period, set_up_cube,
-    set_up_spot_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_ResamplePercentiles.py]: fix

File: improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_utilities.py [hunk 8]
```diff
@@ -47,10 +47,10 @@
     insert_lower_and_upper_endpoint_to_1d_array,
     restore_non_probabilistic_dimensions)
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period, set_up_cube,
-    set_up_probability_above_threshold_temperature_cube,
-    set_up_spot_temperature_cube, set_up_temperature_cube)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_utilities.py]: fix

File: improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py [hunk 9]
```diff
@@ -45,8 +45,8 @@
 from improver.nbhood.nbhood import BaseNeighbourhoodProcessing as NBHood
 from improver.nbhood.nbhood import SquareNeighbourhood
 
-from ...ensemble_calibration.ensemble_calibration.helper_functions import (
-    add_forecast_reference_time_and_forecast_period)
+from ...calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/nbhood/nbhood/test_BaseNeighbourhoodProcessing.py]: fix

File: improver_tests/utilities/test_mathematical_operations.py [hunk 10]
```diff
@@ -39,8 +39,8 @@
 
 from improver.utilities.mathematical_operations import Integration
 
-from ..ensemble_calibration.ensemble_calibration.helper_functions import (
-    set_up_temperature_cube)
+from ..calibration.ensemble_calibration.helper_functions \
```
Comment (fionaRust) [inline improver_tests/utilities/test_mathematical_operations.py]: fix

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/calibration/ensemble_calibration.py  13
- improver_tests/calibration/utilities/test_utilities.py  2
- improver_tests/calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  3
- improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  4
- improver_tests/calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py  2
         

Clones added
============
- improver/calibration/ensemble_calibration.py  2
- improver_tests/calibration/utilities/test_utilities.py  10
- improver_tests/calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py  6
- improver_tests/calibration/ensemble_calibration/test_ContinuousRankedProbabilityScoreMinimisers.py  12
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4840541&bid=16072661)

**Pull request merged**